lightbug_http/header.mojo

```diff
@@ -837,31 +851,31 @@ struct headerScanner:
             raise Error("Invalid header, did not find a colon")
         
         var jump_to = colon + 1
-        self.set_key(self.b()[:jump_to])
+        self.set_key(self.__b[:colon])
```

Comment:
did you change this from `...[:jump_to])`to `...[:colon])` on purpose? I remember having to set it to jump_to because otherwise it was off but maybe now with the changes it's not needed

Reply:
Related to my comment from above, I changed it to not include the colon, and then we don't have to check for the colon in the parse_header logic